File: WebsiteAppRPG/Core/Entities/Player.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace WebsiteAppRPG.Core.Entities
{
    [Table("Players")]
    public class Player
    {
        [Key]
        [Column("PlayerID")]
        [JsonPropertyName("playerId")]
        public int PlayerID { get; set; }

        [Column("Email")]
        public string Email { get; set; }

        [Column("Name")]
        public string Name { get; set; }

        [Column("Password")]
        public string Password { get; set; }

        [ForeignKey("CharacterID")]
        [Column("CharacterID")]
        [JsonPropertyName("characterId")]
        public int CharacterID { get; set; }

        public Player()
        {
            
        }

        public Player(string email, string name, string password, int characterId)
        {
            Email = email;
            Name = name;
            Password = password;
            CharacterID = characterId;
        }
    }
}

File: WebsiteAppRPG/Frontend/website-app-rpg-frontend/src/models/player.ts
export class Player{

    public playerId: number;

    public email: string;

    public name: string;

    public password: string;

    public characterID: number;

    constructor(playerId: number, email: string, name: string, password: string, characterId: number)
    {
        this.playerId = playerId;
        this.email = email;
        this.name = name;
        this.password = password;
        this.characterID = characterId;
    }
}
